### PR TITLE
Port of m1 tests for guest checkout events eci 53

### PR DESCRIPTION
**Pull request by tdieterich-drip**
Addresses [ECI-53](https://dripcom.atlassian.net/browse/ECI-53)

We now listen for `sales_order_save_after` in addition to `sales_order_save_commit_after`, reacting to the first only when it's a guest checkout, and the later when it's a registered customer checkout.

### CODE REVIEW

File: devtools_m2/cypress/integration/CustomerCartInteractions/steps.js [hunk 1]
```diff
@@ -14,6 +14,8 @@ When('I create an account', function() {
     cy.get('input[name="password_confirmation"]').type('blahblah123!!!')
     cy.contains('Create an Account').click()
   })
+  cy.get('.message-success > div').contains('Thank you for registering with')
+  cy.visit('/lib/web/blank.html')
```
Comment (tdieterich-drip) [inline devtools_m2/cypress/integration/CustomerCartInteractions/steps.js]: ☝️ Not really part of the issue, but I found and edge case where waiting for the registration process to complete helped the stability the test platform.
